### PR TITLE
correctly calculate read quorum based on the available fileInfo

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
correctly calculate read quorum based on the available fileInfo

## Motivation and Context
The current usage of assuming `default` parity of `4` is not correct
for all objects stored on MinIO, objects in .minio.sys have maximum
parity, healing won't trigger on these objects due to incorrect
verification of quorum.

## How to test this PR?
Create a canned policy for example 

```
~ minio server /tmp/xl{1...16}
~ mc admin policy add alias/ custom-policy ./policy.json
```

```
~ rm -rf /tmp/xl{1..8}/.minio.sys/config/iam/policies/custom-policy/policy.json
```

```
~ mc admin heal -r alias/
```

Would return an error for `custom-policy/policy.json`  due to incorrect
quorum calculation.

This PR fixes this behavior.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [x] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-healing.go [hunk 1]
```diff
@@ -306,18 +304,27 @@ func (er erasureObjects) healObject(ctx context.Context, bucket string, object s
...
+	}
 
-	if _, err = getLatestFileInfo(ctx, partsMetadata, errs); err != nil {
+	readQuorum, _, err := objectQuorumFromMeta(ctx, partsMetadata, errs, er.defaultParityCount)
+	if err != nil {
 		return er.purgeObjectDangling(ctx, bucket, object, versionID, partsMetadata, errs, []error{}, opts)
 	}
 
+	result.ParityBlocks = result.DiskCount - readQuorum
+	result.DataBlocks = result.DiskCount - result.ParityBlocks
```
Comment (krishnasrinivas) [inline cmd/erasure-healing.go]: you could just do:
```
result.DataBlocks = readQuorum
```

Comment (harshavardhana) [reply cmd/erasure-healing.go]: Sure @krishnasrinivas

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by krishnasrinivas — APPROVED** [review]

**Pull request merged**